Check Bukkit API version on load.

bigpresh opened the pull request:
This is for #119.

Need to consider if this is the best way.

Need also to consider if the latest GP builds will be able to work on 1.11 and 1.12 or just 1.12 only.

bigpresh committed:
Check Bukkit API version on load.

RoboMWM commented on src/me/ryanhamshire/GriefPrevention/GriefPrevention.java:
```diff
@@ -256,6 +256,17 @@ public void onEnable()
 		AddLogEntry("Finished loading configuration.");
+
+                // Check we can run with this Minecraft version
+                if (!Bukkit.getVersion().startsWith("1.12")) {
```
I'd like getServer() but there's already a lot of static usage in this I guess... Also, the current and upcoming version is compatible with 1.11.2 as well.

Might need to mark this with a `TODO` so we remember to change it or something

RoboMWM commented on src/me/ryanhamshire/GriefPrevention/GriefPrevention.java:
```diff
@@ -256,6 +256,17 @@ public void onEnable()
+                            + Bukkit.getVersion()
+                            + " - download an appropriate GP version"
+                    );
+                    return;
```
If this is a warning, why are we returning?

bigpresh replied:
I think I did  this as a "it isn't supported, don't load", but that might be a bit excessive - warning that it might not work but letting them try is probably more user-friendly - in a lot of cases it will still work fine with a slightly newer Minecraft version that doesn't contain major changes, so it would be a bit nasty to refuse to try.

RoboMWM reviewed (changes requested):
Probably best to _also_ send a message via a scheduled task so it appears right after the server finishes loading all plugins. Other than that, I guess it's okay.

bigpresh committed:
Refactor, support multiple compatible versions.

bigpresh committed:
Just warn, no need to bail

bigpresh committed:
Log the version warning again in a bit

As per
https://github.com/TechFortress/GriefPrevention/pull/154#issuecomment-323302511
... log the unsupported version warning again after a delay so it appears after
plugin loading is complete.

(Not yet tested)

bigpresh committed:
Pass the instance, not this.plugin

RoboMWM commented on src/me/ryanhamshire/GriefPrevention/GriefPrevention.java:
```diff
@@ -256,6 +257,42 @@ public void onEnable()
+
+                // Check we can run with this Minecraft version
+                final String[] supportedVersions = {
+                   "1.11.2", "1.12", "1.12.1"
```
Aside from the obvious conflict due to the maven refactor PR I just pulled, this needs to support future versions, or at least only give an info message (i.e. using newer MC versions causes this to not scream as much as using older ones).

RoboMWM replied:
Especially since it currently is running and functioning on the spigot test server, which is running private 1.13 spigot snapshot builds
